### PR TITLE
Fix various list bugs

### PR DESCRIPTION
**Pull request by srawlins**
This fixes various list bugs, bringing our CommonMark compliance up from 465 (75.5%) to 483 (78.4%).

* Nested lists were only enabled with 3 spaces or a tab. Now you need to indent as far as the content of the parent item.
* Blank list items are allowed.
* Two blank lines ends a list.
* List items that start with an indented code block work ok.

### CODE REVIEW

**Comment by srawlins** [comment]
CC @kevmoo @munificent

File: lib/src/block_parser.dart [hunk 1]
```diff
@@ -512,17 +517,57 @@ abstract class ListSyntax extends BlockSyntax {
...
+
     while (!parser.isDone) {
       if (tryMatch(_emptyPattern)) {
+        if (_emptyPattern.firstMatch(parser.next ?? '') != null) {
+          // Two blank lines ends a list.
+          break;
+        }
         // Add a blank line to the current list item.
         childLines.add('');
+      } else if (tryMatch(new RegExp('^$indent(.*)'))) {
```
Comment (munificent) [inline lib/src/block_parser.dart]: Can you just do `startsWith(indent)` here?

Comment (srawlins) [inline lib/src/block_parser.dart]: Done. Much cheaper.

**Comment by kevmoo** [comment]
Rebase against latest + run the update script?

**Comment by srawlins** [comment]
@kevmoo Good idea! Done.

**Comment by kevmoo** [comment]
@srawlins check travis?

**Comment by srawlins** [comment]
@kevmoo good catch. done!

**Pull request merged**